state_machine/tests/test_state_machine.py

```diff
@@ -0,0 +1,368 @@
+import unittest
+import sys
+
+sys.path.insert(0, './state_machine/frame')
+
+from lib.state_machine_utils import validate_config  # noqa: E402
+
+basicTM = {
+    't1': True,
+    't2': True,
+    't3': True,
+}
+
+basicTFM = {
+    'tf1': True,
+    'tf2': True,
+}
+
+
+tests = [
```

Comment:
This can be a more long term change (as it's about readability rather than functionality), but it may make sense to define variables to hold certain information instead of having multiple nested dictionaries. Ex. you can create a "Normal" dictionary that holds the following information:

{
                'Tasks': {
                    't1': {
                        'Interval': 10,
                        'Priority': 3,
                        'ScheduleLater': False
                    },
                    't2': {
                        'Interval': 10,
                        'Priority': 3,
                        'ScheduleLater': False
                    },
                    't3': {
                        'Interval': 10,
                        'Priority': 3,
                        'ScheduleLater': False
                    },
                },
                'StepsTo': ['LowPower', 'DeTumble'],
                'EnterFunctions': ['tf1'],
                'ExitFunctions': ['tf2'],
            },

and then pass that variable name into your config dictionary instead of writing it all at once (as the nested dictionaries and indents are a little difficult to keep track of)

Reply:
Unfortunately there isn't much duplication of codeblocks like this, this is generally the area that I am slightly tweaking to make test cases.